### PR TITLE
handle iso mount error on RDU2 Ampere ARM servers with retry

### PR DESCRIPTION
**Pull request by bmanzari**
prow step code handles iso mount incorrectly not catching an exception

the server is powered On but not having any bootable drive ready turns off after 3/4 minutes, install fails because the node never joins

### CODE REVIEW

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by bmanzari** [comment]
/assign @aleskandro 

test job https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/41956/rehearse-41956-periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7/1687362786805420032#1:build-log.txt%3A339

File: ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh [hunk 1]
```diff
@@ -40,7 +40,48 @@ bmc_password = sys.argv[3]
...
+            print("Task state: %s " % task.dict.get("TaskState"))
+            print("Task status: %s" % task.status)
+            #print("Task dict: %s" % task.dict)
+            retry_time = task.retry_after
+            time.sleep(retry_time if retry_time else 5)
+            if (task.dict.get("TaskState") in ("Exception")):
+               if (retry_counter < 6):
+                    print("Exception in %s task Mount Remote, retrying" % bmc_address)
+                    retry_counter=retry_counter+1
+                    redfish_mount_remote(context=context)
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh:75]: hi @bmanzari, thanks for the PR. If we could avoid the recursion by using a for loop, I'd prefer it. If you still prefer to use recursion, I'd let the `retry_counter` var be a local parameter of the function.

But in general, it looks good to me. Once you conclude testing and remove the WIP, I'm ok to approve it.

Comment (bmanzari) [reply ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh]: now using a loop and local vars, let me know

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by bmanzari** [comment]
test job recovering 2 out of 3 iso mounting failures (search for 'Task state: Exception')

https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/41956/rehearse-41956-periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7/1689559861789986816/artifacts/baremetal-compact-agent-ipv4-static-connected-p1-f7/baremetal-lab-agent-install/build-log.txt

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7

**Comment by bmanzari** [comment]
test job recovering 1 out of 3 iso mounting failures (host 032, search for 'Task state: Exception')

https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/41956/rehearse-41956-periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-compact-agent-ipv4-static-connected-p1-f7/1695368499787993088/artifacts/baremetal-compact-agent-ipv4-static-connected-p1-f7/baremetal-lab-agent-install/build-log.txt

**Pull request merged**